pallets/snarcos/src/lib.rs

```diff
@@ -254,29 +254,53 @@ pub mod pallet {
             proof: Vec<u8>,
             public_input: Vec<u8>,
         ) -> Result<(), (Error<T>, Option<Weight>)> {
+            let data_length_limit = T::MaximumDataLength::get() as usize;
+            let data_length_excess = proof.len().saturating_sub(data_length_limit)
+                + public_input.len().saturating_sub(data_length_limit);
             ensure!(
-                proof.len() <= T::MaximumDataLength::get() as usize
-                    && public_input.len() <= T::MaximumDataLength::get() as usize,
-                (Error::<T>::DataTooLong, None)
+                data_length_excess == 0,
+                (
+                    Error::<T>::DataTooLong,
+                    Some(T::WeightInfo::verify_data_too_long(
+                        data_length_excess as u32
```

Comment:
Have we discussed this ever before and agreed that we will charge the exact amount of gas that is equal to the exceess data length?

Reply:
We will charge `T::WeightInfo::verify_data_too_long(data_length_excess as u32)`, where `T::WeightInfo::verify_data_too_long(x)` tells us how much weight is consumed for handling extrinsic with arguments that has `x` bytes over limit.

Reply:
Why not charge with `proof.len()` and `public_input.len()` though? Again, cost is constant so it probably doesn't matter.

Reply:
OK, I think I see the reason now. If `x` was valid (i.e. not exceeding the limit) then the cost of this operation would be different so there's no real reason to use _that_ value here instead.